diff --git a/node_modules/@quasar/app/bin/quasar b/node_modules/@quasar/app/bin/quasar
index ef5dc7f..01a07ac 100644
--- a/node_modules/@quasar/app/bin/quasar
+++ b/node_modules/@quasar/app/bin/quasar
@@ -15,7 +15,8 @@ const commands = [
   'info',
   'new',
   'test',
-  'help'
+  'help',
+  'src'
 ]
 
 let cmd = process.argv[2]
diff --git a/node_modules/@quasar/app/bin/quasar-build b/node_modules/@quasar/app/bin/quasar-build
index 7b9cf2d..95b7fce 100644
--- a/node_modules/@quasar/app/bin/quasar-build
+++ b/node_modules/@quasar/app/bin/quasar-build
@@ -259,7 +259,10 @@ async function build () {
 
     // free up memory
     webpackData = void 0
-
+    if(argv.src) {
+      log(`skip yarn install --production and electron-packager`)
+      return
+    }
     finalizeBuild(argv.mode, ctx, quasarConfFile).then(async () => {
       outputFolder = argv.mode === 'cordova'
         ? path.join(outputFolder, '..')
diff --git a/node_modules/@quasar/app/lib/electron/index.js b/node_modules/@quasar/app/lib/electron/index.js
index 1b5152e..14399c7 100644
--- a/node_modules/@quasar/app/lib/electron/index.js
+++ b/node_modules/@quasar/app/lib/electron/index.js
@@ -14,6 +14,7 @@ class ElectronRunner {
   constructor () {
     this.pid = 0
     this.watcher = null
+    this.e2e_test_mode = false
   }
 
   init () {}
@@ -21,6 +22,11 @@ class ElectronRunner {
   async run (quasarConfFile, argv) {
     const url = quasarConfFile.quasarConf.build.APP_URL
 
+    if(argv.e2e) {
+      log('Building main Electron for E2E testing...')
+      this.e2e_test_mode = true
+    }
+    
     if (this.pid) {
       if (this.url !== url) {
         await this.stop()
@@ -60,8 +66,10 @@ class ElectronRunner {
         }
 
         await this.__stopElectron()
+        
         this.__startElectron(argv._)
 
+
         resolve()
       })
 
@@ -184,6 +192,12 @@ class ElectronRunner {
   }
 
   __startElectron (extraParams) {
+
+    if(this.e2e_test_mode) {
+      log(`Ignoring boot up Electron for e2e-testing...`)
+      return
+    }
+
     log(`Booting up Electron process...`)
     this.pid = spawn(
       getPackage('electron'),
