diff --git a/node_modules/electron-installer-debian/resources/control.ejs b/node_modules/electron-installer-debian/resources/control.ejs
index d6ef44e..df60c7f 100644
--- a/node_modules/electron-installer-debian/resources/control.ejs
+++ b/node_modules/electron-installer-debian/resources/control.ejs
@@ -10,6 +10,9 @@
 <% } %><% if (preDepends && preDepends.length) { %>Pre-Depends: <%= preDepends.join(', ') %>
 <% } %><% if (size) { %>Installed-Size: <%= size %>
 <% } %><% if (maintainer) { %>Maintainer: <%= maintainer %>
+<% } %><% if (name) { %>Provides: <%= name %>
+<% } %><% if (name) { %>Conflicts: <%= name %>
+<% } %><% if (name) { %>Replaces: <%= name %>
 <% } %><% if (homepage) { %>Homepage: <%= homepage %>
 <% } %><% if (description) { %>Description: <%= description %>
 <% } %><% if (productDescription) { %><%= productDescription %><% } %>
